src/main/java/com/google/sps/servlets/PlaceGuideServlet.java

```diff
@@ -0,0 +1,111 @@
+  public static final String DESC_INPUT = "desc";
+  public static final String LENGTH_INPUT = "length";
+  public static final String IMG_KEY_INPUT = "imgKey";
+  public static final String PLACE_GUIDE_TYPE_PARAMETER = "placeGuideType";
+
+  private final PlaceGuideRepository placeGuideRepository = PlaceGuideRepositoryFactory
+                                                .getPlaceGuideRepository(RepositoryType.DATASTORE);
+  private final String userId = UserServiceFactory.getUserService().getCurrentUser().getUserId();
+
+  /**
+   * Saves the recently submitted place guide data.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PlaceGuide placeGuide = getPlaceGuideFromRequest(request);
+    placeGuideRepository.savePlaceGuide(userId, placeGuide);
+    response.sendRedirect("/createPlaceGuide.html");
+  }
+
+  /**
+   * Returns the data of the placeguide(s) asked by the user who is currently logged in.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String placeGuideType = request.getParameter(PLACE_GUIDE_TYPE_PARAMETER);
+    List<PlaceGuide> placeGuides = null;
+    switch(placeGuideType) {
+      case ALL:
+        placeGuides = placeGuideRepository.getAllPlaceGuides();
+        break;
+      case CREATED_ALL:
+        placeGuides = placeGuideRepository.getCreatedPlaceGuides(userId);
+        break;
+      case CREATED_PUBLIC:
+        placeGuides = placeGuideRepository.getCreatedPublicPlaceGuides(userId);
+        break;
+      case CREATED_PRIVATE:
+        placeGuides = placeGuideRepository.getCreatedPrivatePlaceGuides(userId);
+        break;
+      default:
+        throw new IllegalStateException("Place Guide type does not exist!");
+    }
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJsonUsingGson(placeGuides));
+  }
+
+  private PlaceGuide getPlaceGuideFromRequest(HttpServletRequest request) {
+    String name = request.getParameter(NAME_INPUT);
+    String audioKey = request.getParameter(AUDIO_KEY_INPUT); // Get from Blobstore.
+    String placeId = request.getParameter(PLACE_ID_INPUT);
+    float latitude = Float.parseFloat(request.getParameter(LATITUDE_INPUT));
+    float longitude = Float.parseFloat(request.getParameter(LONGITUDE_INPUT));
+    GeoPt coord = new GeoPt(latitude, longitude);
+    PlaceGuide.Builder newPlaceGuideBuilder = new PlaceGuide.Builder(name, audioKey, userId, 
+                                                                     placeId, coord);
+    String publicPlaceGuideStringValue = request.getParameter(IS_PUBLIC_INPUT);
+    if (publicPlaceGuideStringValue.equals(IS_PUBLIC_INPUT_VALUE)) {
+      newPlaceGuideBuilder.setPlaceGuideToPublic(true);
+    }
+    String length = request.getParameter(LENGTH_INPUT);
+    if (!length.isEmpty()) {
+      newPlaceGuideBuilder.setLength(Integer.parseInt(length));
```

Comment:
`length` is a long in PlaceGuide, consider `Long.parseLong()`

Reply:
modified.